docs/06-contributors/999-rfcs/000-stories/story-15.md

```diff
@@ -0,0 +1,186 @@
+  inflight remove(id: str): void; 
+  inflight find(r: str): Array<str>;
+  inflight set_status(id: str, status: Status): str;
+  inflight set_estimation(id: str, estimation: duration): str;
+}
+
+resource TaskList implementes ITaskList {
+  _redis: redis.Redis;
+  init() {
+    this._redis = new redis.Redis();
+    this.inflights.add("get", ref: "this._redis", op: "GET" );
+    this.inflights.add("_add", ref: "this._redis", op: ["SET", "SADD"]);
+    // notice I am setting explicit permissions on this
+    this.inflights.add("add", ref: "this", op: "_add");
+    // is this the right synatx for multiple ops? 
+    this.inflights.add("set_status", ref: "this", op: ["_add", "get"]); 
+    this.inflights.add("set_estimation", ref: "this", op: ["_add", "get"]); 
+
+    // TODO add more permissions for remove, find
+  }
+
+  inflight get(id: str): Json {
+     return Json.parse(this._redis.GET(id));
+  }
+  
+  inflight _add(id: str, j: Json): str {
+    this._redis.SET(id , Json.format(j));
+    this._redis.SADD("todo", id);
+    return id;
+  } 
+  
+  inflight add(title: str): str {
+    // PLACEHOLDER - how does untyped works with numeric operations
+    let id = "${js.Math.floor(js.Math.random() * 100000000000)}";
+    let j = Json { 
+      title: title, 
+      status: Status.Uncompleted
+    };
+    print("adding task ${id} with data: ${j}"); 
+    return this_add(id, js);
+  }
+
+  inflight remove(id: str) {
+    print("removing task ${id}");
+    this._redis.DEL(id);
+  }
+
+  // PLACEHOLDER - having an untyped type
+  inflight find(term: str): Array<str> { 
+    let r = new js.RegExp(term);
+    let result = MutArray<str>[]; 
+    let ids = this._redis.SMEMBERS("todo");
+    for id in ids {
+      let j = Json.parse(this._redis.GET(id));
+      // Notice that there is autocasting from untyped to bool here 
+      if r.test(j.title) {
+        result.push(id);
+      }
+    }
+    return result.copy();
+  }
+
+  inflight set_status(id: str, status: Status): str {
+    let j = Json.clone_mut(this.get(id));
+    j.status = status;
+    this._add(id, Json.clone(j));
+    return id;
+  }
+
+  inflight set_estimation(id: str, estimation: duration): str {
+    let j = Json.clone_mut(this.get(id));
+    j.effort_estimation = estimation;
+    this._add(id, j);
+    return id;
+  }
+}
+
+resource TaskListApi {
+  api: cloud.Api;
+  task_list: ITaskList;
+  init(task_list: ITaskList) {
+    this.task_list = task_list;
+    this.api = new cloud.Api();
+    
+    // TODO add this.put
+    
+    this.api.post("/tasks", inflight (req: cloud. Api.ApiRequest): cloud.ApiResponse => {
+      let var title = str.from_json(req.body.title);
+      // Easter Egg - if you add a todo with the single word "random" as the title, 
+      //              the system will fetch a random task from the internet
+      if title == "random" {
+        // PLACEHOLDER - can I cast an untyped ?
+        let random_task = axios.get('https://www.boredapi.com/api/activity');
```

Comment:
@eladb , following the comments from the team, I understand that all function calls (in inflight context), and not just inflight method on resources are implicitly using await

Is that the intent?